// @uiw/react-markdown-editor/cjs/commands/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "bold", {
  enumerable: true,
  get: function get() {
    return _bold.bold;
  }
});
Object.defineProperty(exports, "code", {
  enumerable: true,
  get: function get() {
    return _code.code;
  }
});
Object.defineProperty(exports, "codeBlock", {
  enumerable: true,
  get: function get() {
    return _code.codeBlock;
  }
});
exports.defaultCommands = void 0;
Object.defineProperty(exports, "fullscreen", {
  enumerable: true,
  get: function get() {
    return _fullscreen.fullscreen;
  }
});
exports.getModeCommands = exports.getCommands = void 0;
Object.defineProperty(exports, "header", {
  enumerable: true,
  get: function get() {
    return _header.header;
  }
});
Object.defineProperty(exports, "image", {
  enumerable: true,
  get: function get() {
    return _image.image;
  }
});
Object.defineProperty(exports, "italic", {
  enumerable: true,
  get: function get() {
    return _italic.italic;
  }
});
Object.defineProperty(exports, "link", {
  enumerable: true,
  get: function get() {
    return _link.link;
  }
});
Object.defineProperty(exports, "olist", {
  enumerable: true,
  get: function get() {
    return _olist.olist;
  }
});
Object.defineProperty(exports, "preview", {
  enumerable: true,
  get: function get() {
    return _preview.preview;
  }
});
Object.defineProperty(exports, "quote", {
  enumerable: true,
  get: function get() {
    return _quote.quote;
  }
});
Object.defineProperty(exports, "redo", {
  enumerable: true,
  get: function get() {
    return _redo.redo;
  }
});
Object.defineProperty(exports, "strike", {
  enumerable: true,
  get: function get() {
    return _strike.strike;
  }
});
Object.defineProperty(exports, "todo", {
  enumerable: true,
  get: function get() {
    return _todo.todo;
  }
});
Object.defineProperty(exports, "ulist", {
  enumerable: true,
  get: function get() {
    return _ulist.ulist;
  }
});
Object.defineProperty(exports, "underline", {
  enumerable: true,
  get: function get() {
    return _underline.underline;
  }
});
Object.defineProperty(exports, "undo", {
  enumerable: true,
  get: function get() {
    return _undo.undo;
  }
});
var _bold = require("./bold");
var _code = require("./code");
var _italic = require("./italic");
var _header = require("./header");
var _strike = require("./strike");
var _underline = require("./underline");
var _olist = require("./olist");
var _ulist = require("./ulist");
var _quote = require("./quote");
var _link = require("./link");
var _todo = require("./todo");
var _image = require("./image");
var _redo = require("./redo");
var _undo = require("./undo");
var _fullscreen = require("./fullscreen");
var _preview = require("./preview");
var defaultCommands = exports.defaultCommands = {
  undo: _undo.undo,
  redo: _redo.redo,
  bold: _bold.bold,
  italic: _italic.italic,
  header: _header.header,
  strike: _strike.strike,
  underline: _underline.underline,
  quote: _quote.quote,
  olist: _olist.olist,
  ulist: _ulist.ulist,
  todo: _todo.todo,
  link: _link.link,
  image: _image.image,
  code: _code.code,
  codeBlock: _code.codeBlock,
  fullscreen: _fullscreen.fullscreen,
  preview: _preview.preview
};
var getCommands = exports.getCommands = function getCommands() {
  return Object.keys(defaultCommands).filter(function (key) {
    return !/^(fullscreen|preview)/.test(key);
  }).map(function (key) {
    return defaultCommands[key];
  });
};
var getModeCommands = exports.getModeCommands = function getModeCommands() {
  return [_preview.preview, _fullscreen.fullscreen];
};
